src/classes/CRLP_Batch_Base.cls

```diff
@@ -433,24 +433,64 @@ public abstract class CRLP_Batch_Base {
                 'FROM ' + detailObjectType + ' ' +
                 'WHERE ';
 
-        // Set to CloseDate for testing, SystemModstamp for prod.
-        String whereClauseDateField = Test.isRunningTest() ? 'CloseDate' : 'SystemModstamp';
-        whereClause += whereClauseDateField + ' ';
-
-        if (maxDaysAgo != null && maxYearsAgo != null) {
-            if ((maxDaysAgo / 365) > maxYearsAgo) {
+        if (hasUserDefinedInnerJoin) {
+            whereClause += userDefinedLastNDaysField + ' >= LAST_N_DAYS:' + userDefinedLastNDays;
+        } else {
+            // Set to CloseDate for testing, SystemModstamp for prod.
+            String whereClauseDateField = Test.isRunningTest() ? 'CloseDate' : 'SystemModstamp';
+            whereClause += whereClauseDateField + ' ';
+
+            if (maxDaysAgo != null && maxYearsAgo != null) {
+                if ((maxDaysAgo / 365) > maxYearsAgo) {
+                    whereClause += '>= LAST_N_DAYS:' + maxDaysAgo;
+                } else {
+                    whereClause += '>= LAST_N_YEARS:' + maxYearsAgo;
+                }
+            } else if (maxDaysAgo != null) {
                 whereClause += '>= LAST_N_DAYS:' + maxDaysAgo;
-            } else {
+            } else if (maxYearsAgo != null) {
                 whereClause += '>= LAST_N_YEARS:' + maxYearsAgo;
             }
-        } else if (maxDaysAgo != null) {
-            whereClause += '>= LAST_N_DAYS:' + maxDaysAgo;
-        } else if (maxYearsAgo != null) {
-            whereClause += '>= LAST_N_YEARS:' + maxYearsAgo;
         }
         return whereClause += ')';
     }
 
+    private Boolean hasUserDefinedInnerJoin {
+        get {
+            return userDefinedLastNDaysField != null && userDefinedLastNDays != null;
```

Comment:
@kslewis I was originally thinking that the customer can able to define or or both of these fields independently rather than require them to define both, but I think this fine as is for the MVP